autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/passwordgeneration/AutofillUseGeneratedPasswordDialogFragment.kt

```diff
@@ -202,28 +206,37 @@ class AutofillUseGeneratedPasswordDialogFragment : BottomSheetDialogFragment(),
         object GeneratedPasswordAccepted : DialogEvent
     }
 
-    private fun getOriginalUrl() = arguments?.getString(UseGeneratedPasswordDialog.KEY_URL)!!
+    private fun getAutofillUrlRequest() = arguments?.safeGetParcelable<AutofillUrlRequest>(UseGeneratedPasswordDialog.KEY_URL)!!
     private fun getUsername() = arguments?.getString(UseGeneratedPasswordDialog.KEY_USERNAME)
     private fun getGeneratedPassword() = arguments?.getString(UseGeneratedPasswordDialog.KEY_PASSWORD)!!
     private fun getTabId() = arguments?.getString(UseGeneratedPasswordDialog.KEY_TAB_ID)!!
 
     companion object {
 
         fun instance(
-            url: String,
+            autofillUrlRequest: AutofillUrlRequest,
             username: String?,
             generatedPassword: String,
             tabId: String,
         ): AutofillUseGeneratedPasswordDialogFragment {
             val fragment = AutofillUseGeneratedPasswordDialogFragment()
             fragment.arguments =
                 Bundle().also {
-                    it.putString(UseGeneratedPasswordDialog.KEY_URL, url)
+                    it.putParcelable(UseGeneratedPasswordDialog.KEY_URL, autofillUrlRequest)
                     it.putString(UseGeneratedPasswordDialog.KEY_USERNAME, username)
                     it.putString(UseGeneratedPasswordDialog.KEY_PASSWORD, generatedPassword)
                     it.putString(UseGeneratedPasswordDialog.KEY_TAB_ID, tabId)
                 }
             return fragment
         }
     }
+
+    @Suppress("DEPRECATION")
+    @SuppressLint("NewApi")
+    private inline fun <reified T : Parcelable> Bundle.safeGetParcelable(key: String) =
```

Comment:
Yeah, it's annoying copying this around. I did have a brief look before at a nice way to put it somewhere common and failed (though I can't remember why now). I can have another look to see if it can be tidied up.

Reply:
I can tidy this up... will push up the new code tomorrow morning